105-runtime-metrics-ja.tex

```diff
@@ -69,43 +88,80 @@ \subsection{Memory}
  {ets,305920}]
 \end{VerbatimEshell}
 
-This requires some explaining.
+%This requires some explaining.
+これは説明が必要ですね。
 
-First of all, all the values returned are in bytes, and they represent memory \emph{allocated} (memory actively used by the Erlang VM, not the memory set aside by the operating system for the Erlang VM). It will sooner or later look much smaller than what the operating system reports.
+%First of all, all the values returned are in bytes, and they represent memory \emph{allocated} (memory actively used by the Erlang VM, not the memory set aside by the operating system for the Erlang VM).
+まず第一に、すべての返り値はbyte単位の値となっていて、メモリの\emph{割り当てられた}量(Erlang VMが実際に使用しているメモリ量のことで、OSがErlang VMのために確保した量ではありません)を表します。
+%It will sooner or later look much smaller than what the operating system reports.
+遅かれ早かれ、OSが示す値よりはるかに少ない値だとわかるはずです。
 
-The \expression{total} field contains the sum of the memory used for \expression{processes} and \expression{system} (which is incomplete, unless the VM is instrumented!). \expression{processes} is the memory used by Erlang processes, their stacks and heaps. \expression{system} is the rest: memory used by ETS tables, atoms in the VM, refc binaries\footnote{See Section \ref{sec:binaries}}, and some of the hidden data I mentioned was missing.
+%The \expression{total} field contains the sum of the memory used for \expression{processes} and \expression{system} (which is incomplete, unless the VM is instrumented!).
+\expression{total}の項には、\expression{processes} と \expression{system} で利用してるメモリ量の和が入ります(VMを備えない限り、完璧な和にはなりません！)。
+%\expression{processes} is the memory used by Erlang processes, their stacks and heaps.
+\expression{processes}の項は、Erlangプロセスと、スタックやヒープとして使用されるメモリ量です。 % TODO: their = processesだと思うんですがあまり訳に反映されてないような
+i%\expression{system} is the rest: memory used by ETS tables, atoms in the VM, refc binaries\footnote{See Section \ref{sec:binaries}}, and some of the hidden data I mentioned was missing.
+\expression{system}の項は残りです。つまり、ETSテーブルや、VMの中で使われるアトムや、refcバイナリ\footnote{リファレンスカウントされたバイナリ(reference-counted binary)。プロセスヒープ外に保持した実データと、プロセスヒープ内の参照数をカウントするオブジェクトから構成されるバイナリのこと。\ref{sec:binaries}節参照}、失われた隠しデータなどです。%TODO: hidden data I mentioned was missingがしっくりこない。
```

Comment:
"カンマ and" をどう取るか微妙ですが、ここでは (~ is the rest :... ), and (some ..) という接続で取りたいところです。

先程言及した隠しデータは(system に含まれておらず)失われています、というニュアンス。

ここで、「先程」というのは "which is incomplete, unless the VM is instrumented!" のこと。

ぼくはあまり詳しくないので公式ドキュメントの引用でここは逃げますが ^^;
まず erlang:memory に不完全なことが書いてあります。

> The system value is not complete. Some allocated memory that is to be part of this value is not.
>
> When the emulator is run with instrumentation, the system value is more accurate, but memory directly allocated > for malloc (and friends) is still not part of the system value. Direct calls to malloc are only done from OS-specific > runtime libraries and perhaps from user-implemented Erlang drivers that do not use the memory allocation functions in the driver interface.

   Erlang -- erlang  http://erlang.org/doc/man/erlang.html#memory-0

そんで、"instrument" が erl コマンドのオプション

> -instr (emulator flag)
>
>    Selects an instrumented Erlang runtime system (virtual machine) to run, instead of the ordinary one. When running an instrumented runtime system, some resource usage data can be obtained and analyzed using the instrument module. Functionally, it behaves exactly like an ordinary Erlang runtime system."

   Erlang -- erl   http://erlang.org/doc/man/erl.html

長々と書きましたが、system には見えてないデータが有る、ってことだけ伝わればいいのかなと。